packages/@react-spectrum/color/docs/ColorArea.mdx

```diff
@@ -0,0 +1,341 @@
+import {Label} from '@react-spectrum/label';
+
+<Flex gap="size-300" wrap alignItems="flex-end">
+  <ColorArea
+    aria-label="Background color"
+    defaultValue="hsl(0, 100%, 50%)"
+    xChannel="saturation"
+    yChannel="lightness" />
+  <div>
+    <Label
+      id="hsl-aria-labelledby-id">Background color</Label>
+    <ColorArea
+      aria-labelledby="hsl-aria-labelledby-id"
+      defaultValue="hsl(0, 100%, 50%)"
+      xChannel="saturation"
+      yChannel="lightness" />
+  </div>
+</Flex>
+```
+
+### Accessibility
+
+#### Role description
+
+In order to communicate to a screen reader user that the ColorArea adjusts in two dimensions,
+ColorArea provides an `aria-roledescription`, using the localized string "2D Slider", on each of the
+visually hidden `<input>` elements.
+
+#### Value formatting
+
+The `aria-valuetext` of each `<input>` element within the ColorArea is formatted according to the user's locale automatically.
+It will include the localized color channel name and current value for each channel, with the channel name
+and value that the `<input>` element controls coming before the channel name and value for the adjacent
+`<input>` element. For example, for an RGB ColorArea where the `xChannel` is "blue", the `yChannel`
+is "green", when the current selected color is yellow (`rgb(255, 255, 0)`), the `<input>` representing the
+blue channel will have `aria-valuetext` to announce as "Blue: 0, Green: 255", and the `<input>`
+representing the green channel will have `aria-valuetext` to announce as "Green: 255, Blue: 0".
+
+### Internationalization
+
+For languages that are read right-to-left (e.g. Hebrew and Arabic), the layout of the ColorArea is automatically flipped. As mentioned previously,
+ColorArea automatically uses the current locale to format the `aria-valuetext` for each of the `<input>` element with the channel name and current value.
+
+## Events
+
+ColorArea supports two events: `onChange` and `onChangeEnd`. `onChange` is triggered whenever the ColorArea's handle is dragged, and `onChangeEnd`
+is triggered when the user stops dragging the handle. Both events receive a <TypeLink links={colorTypes.links} type={colorTypes.exports.Color} /> object
+as a parameter.
+
+The example below uses `onChange` and `onChangeEnd` to update two separate elements with the ColorArea's value.
+
+```tsx example
+import {ColorArea} from '@react-spectrum/color';
+import {parseColor} from '@react-stately/color';
+
+function Example() {
+  let [currentValue, setCurrentValue] = React.useState(parseColor('hsl(50, 100%, 50%)'));
+  let [finalValue, setFinalValue] = React.useState(parseColor('hsl(50, 100%, 50%)'));
+
+  return (
+    <div>
+      <ColorArea
+        value={currentValue}
+        onChange={setCurrentValue}
+        onChangeEnd={setFinalValue} />
+      <pre>Current value: {currentValue.toString('hsl')}</pre>
+      <pre>Final value: {finalValue.toString('hsl')}</pre>
+    </div>
+  );
+}
+```
+
+## Creating a color picker
```

Comment:
Ideally we'd have a separate `ColorPicker` page (once we have that component), which would document this. Might be a little strange to have this section on the `ColorArea` page vs any if the other color components.

Reply:
This was added to provide guidance similar to what we do for `ColorSlider`. One can build a fully featured color picker with `ColorSlider` because each slider can represent a single channel. `ColorWheel` is only for the hue channel, so it's easy to explain, but not really useful without `ColorArea`. `ColorArea` is difficult to explain without understanding how the xChannel and yChannel relate to the external control of the zChannel.

Reply:
Bottom line, until we devise a fully functional ColorPicker, which would incorporate text inputs for channel values, a swatch to render the current value, and perhaps a "ColorSwatchListBox" for choosing predefined swatches or favorites, and an eyedropper tool for selecting a color from the screen using a cursor. I think we need to include examples on how to combine ColorArea with other controls to create a ColorPicker.

Within Spectrum, Color wheel, Color slider and Color area all have design usage documentation that shows this:

- https://spectrum.adobe.com/page/color-wheel/#Usage-guidelines
- https://spectrum.adobe.com/page/color-slider/#Usage-guidelines
- https://spectrum.adobe.com/page/color-area/#Usage-guidelines